src/core/main.c

```diff
@@ -1586,6 +1589,10 @@ static int become_shutdown(int objective, int retval) {
         r = watchdog_setup(watchdog_timer);
         watchdog_close(r < 0);
 
+        /* The environment block: */
+
+        env_block = strv_copy(environ);
```

Comment:
hmm, no oom check?

Reply:
hmm, we do no oom checks for the whole block. i guess that might be ok then, we can't do much anyway here